internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -113,3 +196,96 @@ func setCPUSLoadBalancing(c *oci.Container, enable bool, schedDomainDir string)
 
 	return nil
 }
+
+func setIRQLoadBalancing(c *oci.Container, enable bool, irqSmpAffinityFile string) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return errors.Errorf("find container %s CPUs", c.ID())
+	}
+
+	content, err := ioutil.ReadFile(irqSmpAffinityFile)
+	if err != nil {
+		return err
+	}
+	currentIRQSMPSetting := strings.TrimSpace(string(content))
+	newIRQSMPSetting, newIRQBalanceSetting, err := UpdateIRQSmpAffinityMask(lspec.Resources.CPU.Cpus, currentIRQSMPSetting, enable)
+	if err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(irqSmpAffinityFile, []byte(newIRQSMPSetting), 0o644); err != nil {
+		return err
+	}
+	if _, err := exec.LookPath("irqbalance"); err != nil {
+		// irqbalance is not installed, skip the rest; pod should still start, so return nil instead
+		logrus.Warnf("irqbalance binary not found: %v", err)
+		return nil
+	}
+	// run irqbalance in daemon mode, so this won't cause delay
+	cmd := exec.Command("irqbalance", "--oneshot")
```

Comment:
If the irqbalance service already exists, then we may have to update /etc/sysconfig/irqbalance config file with IRQBALANCE_BANNED_CPUS and restart irqbalance service. we could run irqbalance --oneshot command only if the service is not present (This is what i did it here https://github.com/pperiyasamy/irq-smp-balance/blob/main/pkg/irq/util.go#L94). 
Shouldn't it be done this way ?

Reply:
yes, it can be nice, can you please open the PR and we will discuss it

Reply:
yes, and we need to discuss whether/how to recover  /etc/sysconfig/irqbalance to its default if the computer node rebooted

Reply:
One additional challenge that I see with the approach to reconfigure the irqbalance service is its platform dependency. We have seen that Linux distributions like SLES, Ubuntu, RHEL all have slightly different approaches for configuring and managing the daemon, and even within one distribution the way can change between major releases.

Can we find a solution that will work on all relevant platforms? Perhaps add parameters to the cri-o config file to tell, which file to update and how to restart the service?

If not, can the solution be split into a generic part inside cri-o that manages a a file on the host with the wanted banned CPUs, and another platform-specific daemon that reconfigures the host's irqbalance service accordingly?

Reply:
Yes, The irqbalance config would present at /etc/sysconfig/ directory in SLES, CentOS platforms whereas in Ubuntu it's present in /etc/default/ directory. Hope this file is just a source file which takes in the format of  `IRQBALANCE_BANNED_CPUS=<value>`. Of course we could pass the config file path using a new RuntimeConfig parameter and made it available to runtime_handler_hooks through Server.config. 
we could recover the irqbalance config at crio start time and banned mask derived from /proc/irq/default_smp_affinitity. 
Could we take this approach ?